### PR TITLE
Log API error

### PR DESCRIPTION
**Pull request by wqsz7xn**
- [x] Log API error, probably can do this for a few more around but unsure which ones are important enough to log
- https://github.com/subquery/subql/issues/671

### CODE REVIEW

File: packages/node/src/indexer/api.service.ts [hunk 1]
```diff
@@ -68,9 +71,9 @@ export class ApiService implements OnApplicationShutdown {
       network.genesisHash &&
       network.genesisHash !== this.networkMeta.genesisHash
     ) {
-      throw new Error(
-        `Network genesisHash doesn't match expected genesisHash. expected="${network.genesisHash}" actual="${this.networkMeta.genesisHash}`,
-      );
+      const err = `Network genesisHash doesn't match expected genesisHash. expected="${network.genesisHash}" actual="${this.networkMeta.genesisHash}`;
```
Comment (ianhe8x) [inline packages/node/src/indexer/api.service.ts:74]:
```suggestion
      const err = new Error(`Network genesisHash doesn't match expected genesisHash. expected="${network.genesisHash}" actual="${this.networkMeta.genesisHash}`);
      logger.error(err, err.message);
      throw err;
```

Comment (ianhe8x) [reply packages/node/src/indexer/api.service.ts:74]: This way it logs error's stack when --output-fmt=json is enabled

**Comment by ianhe8x** [comment]
@wqsz7xn failed at code style check

**Comment by wqsz7xn** [comment]
oh my bad

**Comment by wqsz7xn** [comment]
Not sure why the CI tests are failing, not related to this though

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**